check if patients have origin data first

thecristen opened the pull request:
An error was being thrown when a patient had no origin data. This helps prevent that error.

pkmitre commented on app/assets/javascripts/views/patient_bank/patient_bank_view.js.coffee:
```diff
@@ -110,13 +110,15 @@ class Thorax.Views.PatientBankView extends Thorax.Views.BonnieView
   differenceContext: (difference) ->
     origins = @model.get('patients').models.map (p) -> p.get('origin_data')
-    origin_ids = _.compact(origins).map (o) -> o[0].patient_id
+    if origins
```
Won't this always be true? Map should always return at least an empty array, which is truthy.

pkmitre replied:
And, if it is in fact always true, it looks like this code will not have fixed the issue since the rest of the code is essentially the same...

pkmitre commented on Gemfile.lock:
```diff
@@ -323,6 +323,3 @@ DEPENDENCIES
   turn
   uglifier (>= 1.0.3)
-
-BUNDLED WITH
```
Mind taking this change out of the pull request, since it's unrelated?

Also I think we should probably all try to run the same version of bundler so we don't always get changes to this line...

pkmitre commented on app/assets/javascripts/views/patient_bank/patient_bank_view.js.coffee:
```diff
@@ -110,13 +110,15 @@ class Thorax.Views.PatientBankView extends Thorax.Views.BonnieView
 
   differenceContext: (difference) ->
     origins = @model.get('patients').models.map (p) -> p.get('origin_data')
```
I know this is old code, but thorax collections should support map by default, so I think just

    @model.get('patients').map

ought to work.

thecristen commented:
All correct! I've reverted the initial commit and added a change that worked for me - I flatten the arrays because compacting to get rid of zero-length arrays that were contributing to this error.
